featuretools/primitives/options_utils.py

```diff
@@ -0,0 +1,228 @@
+            initialized_option_dict[option_key] = set(option)
+        elif isinstance(option, dict):
+            initialized_option_dict[option_key] = {key: set(option[key]) for key in option}
+    # initialize ignore_entities and ignore_variables to empty sets if not present
+    if 'ignore_variables' not in initialized_option_dict:
+        initialized_option_dict['ignore_variables'] = {}
+    if 'ignore_entities' not in initialized_option_dict:
+        initialized_option_dict['ignore_entities'] = set([])
+    return initialized_option_dict
+
+
+def _variable_filter_generator(options):
+    def passes_ignore_filter(f):
+        return (not isinstance(f, IdentityFeature) or
+                f.entity.id not in options['ignore_variables'] or
+                f.variable.id not in options['ignore_variables'][f.entity.id])
+
+    def passes_include_filter(f):
+        return (not isinstance(f, IdentityFeature) or
+                (f.entity.id in options['include_variables'] and
+                f.variable.id in options['include_variables'][f.entity.id]))
+
+    if 'include_variables' in options:
+        def variable_filter(f):
+            return passes_include_filter(f) or \
+                (f.entity.id not in options['include_variables'] and
+                    passes_ignore_filter(f))
+    # ignore options initialized to set() if not present
+    else:
+        def variable_filter(f):
+            return passes_ignore_filter(f)
+    return variable_filter
+
+
+def _groupby_filter_generator(options):
+    def passes_include_groupby_filter(f):
+        return (isinstance(f, IdentityFeature) and
+                (f.entity.id in options['include_groupby_variables'] and
+                f.variable.id in options['include_groupby_variables'][f.entity.id]))
+
+    def passes_ignore_groupby_filter(f):
+        return (isinstance(f, IdentityFeature) and (
+                f.entity.id not in options['ignore_groupby_variables'] or
+                (f.entity.id in options['ignore_groupby_variables'] and
+                    f.variable.id not in options['ignore_groupby_variables'][f.entity.id])))
+
+    if 'include_groupby_variables' in options and 'ignore_groupby_variables' in options:
+        def groupby_filter(f):
+            return passes_include_groupby_filter(f) or \
+                (f.entity.id not in options['include_groupby_variables'] and
+                    passes_ignore_groupby_filter(f))
+    elif 'include_groupby_variables' in options:
+        def groupby_filter(f):
+            return passes_include_groupby_filter(f) or \
+                f.entity.id not in options['include_groupby_variables']
+    elif 'ignore_groupby_variables' in options:
+        def groupby_filter(f):
+            return passes_ignore_groupby_filter(f)
+    else:
+        def groupby_filter(f):
+            return (isinstance(f, IdentityFeature))
+    return groupby_filter
+
+
+def ignore_entity(options, entity, groupby=False):
```

Comment:
I wonder if there is a better than for this method. perhaps `ignore_entity_for_primitive`, `ignore_entity_for_options`?